### PR TITLE
Modifier and Expression-level preferences

### PR DESCRIPTION
**Pull request by ogolikhin**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -44,3 +44,24 @@ dotnet_style_parentheses_in_arithmetic_binary_operators = never_if_unnecessary:s
...
 dotnet_style_parentheses_in_relational_binary_operators = always_for_clarity:error     #Prefer var v = (a < b) == (c > d); Over var v = a < b == c > d;
+
+# Modifier preferences
+dotnet_style_require_accessibility_modifiers = omit_if_default:silent  #Prefer accessibility modifiers to be specified except if they are the default modifier.
+
+# Expression-level preferences
+dotnet_style_coalesce_expression = true:error                                    #Prefer var v = x ?? y; Over var v = x != null ? x : y;
+dotnet_style_collection_initializer = true:error                                 #Prefer var list = new List<int> { 1, 2, 3 }; Over var list = new List<int>(); list.Add(1);
+dotnet_style_explicit_tuple_names = true:error                                   #Prefer (string name, int age) customer = GetCustomer(); var name = customer.name; Over var name = customer.Item1;
+dotnet_style_namespace_match_folder = true:error
```
Comment (Anton-A-Sakovich) [inline .editorconfig:55]: Hmmm, I've seen this rule in another place and it doesn't seem to be related to expressions... Should it be here? 🤔

File: .editorconfig [hunk 2]
```diff
@@ -44,3 +44,24 @@ dotnet_style_parentheses_in_arithmetic_binary_operators = never_if_unnecessary:s
...
+dotnet_style_operator_placement_when_wrapping = beginning_of_line:error          #Prefer operators like && on new line
+dotnet_style_prefer_auto_properties = true:error                                 #Prefer int Age { get; }
+dotnet_style_prefer_compound_assignment = true:error                             #Prefer x += 1; Over x = x + 1;
+dotnet_style_prefer_conditional_expression_over_assignment = true:silent         #Prefer string s = expr ? "hello" : "world"; But sometimes if-else is more convinient for big portions of code 
+dotnet_style_prefer_conditional_expression_over_return = true:silent             #Prefer return expr ? "hello" : "world" But sometimes if-else is more convinient (e.g. arguments checking)
+dotnet_style_prefer_inferred_anonymous_type_member_names = true:silent           #Prefer var anon = new { age, name }; var anon = new { age = age, name = name };
+dotnet_style_prefer_inferred_tuple_names = true:error                            #Prefer var tuple = (age, name); Over var tuple = (age: age, name: name); But sometimes custom names are more clear
```
Comment (Anton-A-Sakovich) [inline .editorconfig:64]: This line looks confusing to me. The comment to the right says "But sometimes custom names are more clear", as if indicating `silent` severity, but the rule is set to `error` severity. I think that I forgot what severity we finally decided on 😆 .

File: .editorconfig (hunk 1 above)
Comment (ogolikhin) [reply .editorconfig:55]: Yes, it was in the today's PR. Visual Studio put it into expressions section... Not me!)

File: .editorconfig (hunk 2 above)
Comment (ogolikhin) [reply .editorconfig:64]: I guess no error when we set custom name. But need to check it.
